common/flights/src/store_client.rs

```diff
@@ -159,6 +168,43 @@ impl StoreClient {
         anyhow::bail!("invalid response")
     }
 
+    pub async fn scan_partition(
+        &mut self,
+        db_name: String,
+        tbl_name: String,
+        scan_plan: &ScanPlan,
```

Comment:
What about `scan` or `scan_table`?

The subject is a table, not a partition.

BTW on the server-side, the corresponding function name has an extra `s`: `scan_partitions`

Reply:
change method name  "scan_partition" to "scan",  and "read_partition" to "scan_table"?

Oh, wait, u reviewed last night? thanks for that bro. I've force pushed something for the "final touch", including changing the name of method "read" to "read_partition". any suggestion about the now names?

Reply:
what about `scan_table_partitions` and `read_table_partition`? since the former filters table partitions, the latter returns datablocks of a partition.

Reply:
> what about `scan_table_partitions` and `read_table_partition`? since the former filters table partitions, the latter returns datablocks of a partition.

Nice to me.

@BohuTANG @sundy-li Is there any difference between a `datablock` and a `partition` in our code base? 
If no, using a universal term would be better for stargazers.

Reply:
> Oh, wait, u reviewed last night? thanks for that bro. I've force pushed something for the "final touch"

Not that late for me :DDD